### ISSUE TITLE
Slicing plane and clipping frames are in different coordinate systems

### ISSUE THREAD

**Issue by alisterburt** [napari/napari#3874]
## 🐛 Issue
The layer attributes `Layer.experimental_slicing_plane` and `Layer.experimental_clipping_planes` currently take values for the position (and possibly orientation?) in different coordinate systems. The slicing plane/rendering plane is defined in data coordinates whilst the clipping planes are defined in displayed world coordinates

## Expected behavior
Consistency across the two plane types

## Discussion
@brisvag what do we think is the correct path forwards here? I agree that both objects only exist in the scene and are not really 'data attributes' - in that context, scene coordinates makes the most sense. It's often useful to know where these things are in your data so that you can e.g. interrogate your data around that point/orientation. These attributes also (currently) live on the layer which confuses things further.

My feeling is that both should probably be defined in data coordinates whilst they are both layer attributes, I'm happy to be convinced otherwise though and am just opening up discussion

## Additional context
I think the coordinate system of the clipping planes was updated as a side effect of #3494

**Comment by brisvag** [comment]
> I think the coordinate system of the clipping planes was updated as a side effect of #3494

Yes, though it was definitely intended; I just didn't consider it blocking to unify the two (since we clearly don't have exact plans on how to handle this).

> @brisvag what do we think is the correct path forwards here? I agree that both objects only exist in the scene and are not really 'data attributes' - in that context, scene coordinates makes the most sense.

I think ultimately they have to be "objects" in the scene and layers should subscribe to them, so they should use world coordinates.

> It's often useful to know where these things are in your data so that you can e.g. interrogate your data around that point/orientation. These attributes also (currently) live on the layer which confuses things further.

This is the one thing that keeps me from saying "they should all only be in world space". Ideally one should be able to do both, I suppose; luckily, we can do all of that on the napari side.

> My feeling is that both should probably be defined in data coordinates whilst they are both layer attributes, I'm happy to be convinced otherwise though and am just opening up discussion

When I did this change in vispy, I implemented it so [you can easily change coordinate space on volume instantiation](https://github.com/vispy/vispy/blob/26256fdc2574259dd227022fbce0767cae4e244b/vispy/visuals/volume.py#L639). So unifying slicing and clipping so that they both use data space is as easy as adding one argument :)

If most people agree that unification is more important here, I'm ok with getting both into data space for now; however, we should not forget about #3148!

**Comment by alisterburt** [comment]
> I think ultimately they have to be "objects" in the scene and layers should subscribe to them, so they should use world coordinates.

when you say world, are you talking about the nD napari world or the 3D world that the depiction happens in? If depicted world, I agree

> This is the one thing that keeps me from saying "they should all only be in world space". Ideally one should be able to do both, I suppose; luckily, we can do all of that on the napari side.

Yep! Maybe the best solution here is a simple function/method which takes a plane (defined in the 3D scene coordinates) and a layer, returning a plane object which is in the nD data coordinate system of that layer, good documentation will be needed

> When I did this change in vispy, I implemented it so you can easily change coordinate space on volume instantiation. So unifying slicing and clipping so that they both use data space is as easy as adding one argument :)

noice!

> Yes, though it was definitely intended; I just didn't consider it blocking to unify the two (since we clearly don't have exact plans on how to handle this).
> If most people agree that unification is more important here, I'm ok with getting both into data space for now; however, we should not forget about #3148!

Yep! This all feels contingent on the whole data source thing

**Comment by kevinyamauchi** [comment]
Thanks @alisterburt and @brisvag for the discussion. 

> Yes, though it was definitely intended; I just didn't consider it blocking to unify the two (since we clearly don't have exact plans on how to handle this).

I agree that we have some work to do to figure out the best coordinate system for the rendering and clipping planes to live in. However, I do think they should be the same and it would be worth a small patch to unify them. I am curious what others think though.

**Comment by brisvag** [comment]
> when you say world, are you talking about the nD napari world or the 3D world that the depiction happens in? If depicted world, I agree

Right, it's actually vispy scene coordinate I guess!

**Comment by willingc** [comment]
> some work to do to figure out the best coordinate system for the rendering and clipping planes to live in. However, I do think they should be the same and it would be worth a small patch to unify them. I am curious what others think though.

I'm going to relabel this from a bug to future refactor.
